java/src/processing/mode/java/Commander.java

```diff
@@ -145,7 +146,24 @@ public Commander(String[] args) {
         if (!sketchFolder.exists()) {
           complainAndQuit(sketchFolder + " does not exist.", false);
         }
-        File pdeFile = new File(sketchFolder, sketchFolder.getName() + ".pde");
+        
+        // Check for main file in sketch.properties first, then fall back to default
+        File pdeFile = null;
+        try {
+          Settings props = new Settings(new File(sketchFolder, "sketch.properties"));
+          String mainFileName = props.get("main");
+          if (mainFileName != null) {
+            pdeFile = new File(sketchFolder, mainFileName);
+          }
+        } catch (IOException e) {
+          // sketch.properties doesn't exist or couldn't be read, will use default
+        }
+        
+        // Fall back to default naming convention if no custom main file specified
+        if (pdeFile == null || !pdeFile.exists()) {
+          pdeFile = new File(sketchFolder, sketchFolder.getName() + ".pde");
+        }
+        
         if (!pdeFile.exists()) {
           complainAndQuit("Not a valid sketch folder. " + pdeFile + " does not exist.", true);
```

Comment:
@Stefterv Maybe instead of rejecting the folder when folderName.pde is missing, the CLI could handle the following two two cases:

1.	Opening a folder with multiple .pde files but no explicit main file: ask the user whether they want to rename the folder OR choose which .pde should act as the main one, then generate sketch.properties accordingly.
	
2.	Opening a single .pde file inside a folder without an explicit main file: treat that file as the main one and create sketch.properties to match.

That would be closer to the PDE’s behavior. What do you think?